Implemented getMany, removeMany and setMany

MuaviyaImran opened the pull request:
## Created Methods for getMany, setMany and removeMany #5 

Local storage, Session storage, Idb and Cookie adapter supports these methods.

```
setMany([
     { 'name': 'Alizey'},
     { 'person': {name:'Joe',age:20}},
     {'num':[1,2,3] },
     [ 'prop1', 'val1' ], 
     [ 'prop2', 'val2' ]])   // return true
getMany(['name', 'person', 'num','prop1','prop2']) // returns ["Alizey", {name:'Joe',age:20}, [1, 2, 3],'val1','val2']
removeMany(['name', 'person', 'num','prop1','prop2']) // returns true
```
All methods receives array of objects i.e key value pairs, keys must be array and values can be a string, array, objects, and tuples
Note: OPFS adapter has not been included in this PR.

getify commented on src/many.js:
```diff
@@ -0,0 +1,33 @@
+// many.js
+export function setMany(entries, set) {
```
I think the behavior implemented here is a more complicated API than is justified for such a simple library.

I think the only capabilities we should support would be:

```js
setMany([ ["prop1","val1"], ["prop2","val2"] ]);

setMany({ prop1: "val1", prop2: "val2" });
```

Of course, the contents of each value don't have to be strings (as I'm showing above), they can be rich objects/arrays. But we shouldn't reach into those values in anyway, just pass them along to `set()`, which handles its own serialization of the values being set.

IOW, the following should both work the same:

```js
setMany( Object.entries(someObj) );

setMany( someObj );
```

Also, the contract of `set()` is that it returns `true` (or a `Promise<true>`) if the setting succeeded, or throws an exception if it failed to be set (like quota-exceeded). Your logic here ignores a rejected-promise that would come back from `set()` (as opposed to an exception being synchronously thrown).

The example I posted previously, with `Promise.all()`, would "catch" the rejected promise from any of the sets, and therefore propagate that failure outward to the calling code to detect.

SIDE NOTE: there is a "bug" of sorts in the current library implementations, in that `idb`, `opfs`, and `opfs-worker` are all async (as intended), but `cookie`, `local-storage`, and `session-storage` are all sync -- and they should instead be async. So need to fix that for consistency.

getify commented on src/many.js:
```diff
@@ -0,0 +1,33 @@
+	return true;
+}
+
+export async function getMany(names, get) {
```
for consistency, I think `getMany()` and `removeMany()` should still accept objects (as well as arrays), and it just use the keys -- ignoring the values -- from the object.

MuaviyaImran committed:
implemented setMany, and removeMany for local,session, idb and cookie

MuaviyaImran committed:
Converted session, local and cookies to async functions

MuaviyaImran committed:
implemented getMany for local, session, idb and cookie

MuaviyaImran committed:
Added many.js

MuaviyaImran committed:
Removed commented code

MuaviyaImran committed:
Merge branch 'main' into f/enhancement-set-get-remove-many

MuaviyaImran commented:
@getify Done with requested changes

- Simplified logic for many.js
- Converted `cookie, local-storage, and session-storage` to async
- All test cases passed

MuaviyaImran commented:
@getify please review now

getify commented:
thanks so much! fixed it all up and landed the changes. :)

MuaviyaImran commented:
Welcome.. I would be very happy to help in future. @getify
